File: Assets/Inventory/Scripts/Other/ChangeScene.js
#pragma strict

@script AddComponentMenu ("Inventory/Other/ChangeScene")

function Awake () 
{
		var temp = GameObject.Find("Victim");
		if (temp == null) Debug.Log("player not found");
		DontDestroyOnLoad(temp);
		
		if (FindObjectsOfType(GetType()).Length > 1)
         {
             Destroy(gameObject);
         }
}

function Update ()
{
	if (Input.GetKeyDown(KeyCode.T))
	{

		Application.LoadLevel("battle");
	}
}

File: Assets/Scripts/Map/ChangeScene.cs
using UnityEngine;
using System.Collections;

public class ChangeScene : MonoBehaviour {

	public static ChangeScene control;

	// Use this for initialization
	void Awake () {
		if(control == null) {
			control = this;
			DontDestroyOnLoad(this.gameObject);
		}
		else if (control != this) {
			Destroy(gameObject);
		}
	}
	
}
